core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -310,11 +310,11 @@ public synchronized InetSocketAddress getAddress() throws UnavailableException {
         ex = e;
       }
       disconnect();
-      if (retryPolicy.attemptRetry()) {
+      if (retryPolicy.attempt()) {
```

Comment:
Could you move this into the while condition and then do a check on mClosed outside the loop to throw the appropriate exception?

Reply:
Done